addons/purchase/views/purchase_views.xml

```diff
@@ -249,6 +250,7 @@
                                     <field name="qty_received" string="Received" column_invisible="parent.state not in ('purchase', 'done')" readonly="qty_received_method != 'manual'" optional="show"/>
                                     <field name="qty_invoiced" string="Billed" column_invisible="parent.state not in ('purchase', 'done')" optional="show"/>
                                     <field name="product_uom" string="UoM" groups="uom.group_uom"
+                                        context="{'pre_product_uom': product_uom}"
```

Comment:
This introduces different behavior when you change the uom of a purchase order line depending on if you do it through the web client or through the code.